docs/route.md

```diff
@@ -490,9 +490,13 @@ In example 2, since `Layout` is the top-level component in both routes, the DOM
 
 ---
 
-#### Authentication
+#### Redirection
 
-The RouterResolver's `onmatch` hook can be used to run logic before the top level component in a route is initializated. The example below shows how to implement a login wall that prevents users from seeing the `/secret` page unless they login.
+The RouterResolver's `onmatch` hook can be used to run logic before the top level component in a route is initializated. You can use either `m.route.set()` or `window.history.back()` `.forward()`, etc... When redirecting with the `history` API, the `onmatch` hook must return `new Promise(function(){})` to suspend the resolution of the current route. A fresh Promise must be returned every time since recycling a pending Promise leads to memory leaks. `m.route.set()` takes care of cancelling the resolution of the aborted route, there's no need to return a pending Promise in that case.
```

Comment:
I think we hold promises in mentioned example because you pass promise object to console.log. When DevTools are closed, we have 10 Mb limit for memory usage for console and store only N last promises which fit into 10Mb limit. When DevTools is open we send these promises to frontend and hold references to them to make possible further manipulation with logged objects (e.g. you could store logged object as global and then use it in console).
You can clear console by console.clear() from JS or by click on clear button on top left button in console then collect garbage should collect all promises.